`ent schema` panics if any field contains the `DefaultExprs` annotation

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

```console
$ go tool ent schema ./ent/schema --dialect postgres
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8a3fde]

goroutine 1 [running]:
entgo.io/ent/dialect/sql/schema.(*Postgres).Dialect(0x7f44ac6e1a48?)
        <autogenerated>:1 +0x1e
entgo.io/ent/dialect/sql/schema.(*Atlas).atDefault(0xc0004cf9f0, 0xc0003e16c0, 0xc000290070)
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/dialect/sql/schema/atlas.go:1001 +0x198
entgo.io/ent/dialect/sql/schema.(*Atlas).aColumns(0xc0004cf9f0, 0xc0007b8000, 0xc0003e1790)
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/dialect/sql/schema/atlas.go:955 +0x271
entgo.io/ent/dialect/sql/schema.(*Atlas).tables(0xc0004cf9f0, {0xc00011c3b0, 0x1, 0xc00011c2e8?})
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/dialect/sql/schema/atlas.go:868 +0x327
entgo.io/ent/dialect/sql/schema.Dump.(*Atlas).StateReader.func3(...)
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/dialect/sql/schema/atlas.go:530
ariga.io/atlas/sql/migrate.StateReaderFunc.ReadState(...)
        /home/ix/go/pkg/mod/ariga.io/atlas@v0.31.1-0.20250212144724-069be8033e83/sql/migrate/migrate.go:147
entgo.io/ent/dialect/sql/schema.Dump({0xd367c8, 0x11ec740}, {0x7ffc10e5c371, 0x8}, {0x0, 0x0}, {0xc00011c3b0, 0x1, 0x1}, {0x0, ...})
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/dialect/sql/schema/schema.go:624 +0x2bf
entgo.io/ent/cmd/internal/base.SchemaCmd.func1(0xc0003bf208, {0xc0003ca360, 0x1, 0xb9a16a?})
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/cmd/internal/base/base.go:249 +0x292
github.com/spf13/cobra.(*Command).execute(0xc0003bf208, {0xc0003ca2d0, 0x3, 0x3})
        /home/ix/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x843
github.com/spf13/cobra.(*Command).ExecuteC(0xc0003be308)
        /home/ix/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(...)
        /home/ix/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
        /home/ix/go/pkg/mod/entgo.io/ent@v0.14.2/cmd/ent/ent.go:25 +0xc7
```


## Expected Behavior 🤔

Display SQL without error / panic

## Steps to Reproduce 🕹

- reproduce repo: https://github.com/ix64/ent-issue-4337

1. Create schema with `DefaultExprs` annotation
```go
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.Time("created_at").
			Default(time.Now).
			Annotations(entsql.DefaultExprs(map[string]string{
				dialect.Postgres: "NOW()",
			})),
	}
}
```

2. Run 
    - `ent schema ./ent/schema --dialect postgres  --version 17`
    - `ent schema ./ent/schema --dialect mysql --version 8`
    - `ent schema ./ent/schema --dialect sqlite3`



## Your Environment 🌎


| Tech        | Version |
| ----------- | ------- |
| Go          | 1.24.0  |
| Ent         | 0.14.2   |
| Database    | Postgres 17 |
| Driver      | github.com/jackc/pgx/v5/stdlib       |

--- a/dialect/sql/schema/schema.go
+++ b/dialect/sql/schema/schema.go
@@ -594,17 +594,26 @@ type driver struct {
 var drivers = func(v string) map[string]driver {
 	return map[string]driver{
 		entdialect.SQLite: {
-			&SQLite{WithForeignKeys: true},
+			&SQLite{
+				WithForeignKeys: true,
+				Driver:          nopDriver{dialect: entdialect.SQLite},
+			},
 			sqlite.DefaultDiff,
 			sqlite.DefaultPlan,
 		},
 		entdialect.MySQL: {
-			&MySQL{version: v},
+			&MySQL{
+				version: v,
+				Driver:  nopDriver{dialect: entdialect.MySQL},
+			},
 			mysql.DefaultDiff,
 			mysql.DefaultPlan,
 		},
 		entdialect.Postgres: {
-			&Postgres{version: v},
+			&Postgres{
+				version: v,
+				Driver:  nopDriver{dialect: entdialect.Postgres},
+			},
 			postgres.DefaultDiff,
 			postgres.DefaultPlan,
 		},

